src/snowflake/snowpark/_internal/utils.py

```diff
@@ -1924,6 +1925,141 @@ def remove_comments(sql_query: str, uuids: List[str]) -> str:
     )
 
 
+class TimeTravelConfig(NamedTuple):
+    """Configuration for time travel operations."""
+
+    mode: str  # 'at' or 'before'
+    statement: Optional[str] = None
+    offset: Optional[int] = None
+    timestamp: Optional[str] = None
+    timezone: str = "NTZ"
+    stream: Optional[str] = None
+
+
+def validate_and_normalize_time_travel_params(
+    time_travel_mode: Optional[str] = None,
+    statement: Optional[str] = None,
+    offset: Optional[int] = None,
+    timestamp: Optional[Union[str, datetime.datetime]] = None,
+    timezone: Optional[str] = "NTZ",
+    stream: Optional[str] = None,
+) -> Optional[TimeTravelConfig]:
+    """
+    Validates and normalizes time travel parameters.
+    Returns:
+        TimeTravelConfig if time travel is specified, None otherwise.
+    Raises:
+        ValueError: If parameters are invalid.
+    """
+    time_travel_arg_count = sum(
+        arg is not None for arg in (statement, offset, timestamp, stream)
+    )
+
+    # Validate mode
+    if time_travel_mode is None:
+        if time_travel_arg_count == 0:
+            return None
+        else:
+            raise ValueError(
+                "Must specify time travel mode 'at' or 'before' if any other time travel parameter is provided."
+            )
```

Comment:
I'm curious what's the error message if there is a SQL having multiple parameters set?